src/OrderFormAcceptanceTests.Actions/Pages/OrderForm.cs

```diff
@@ -132,7 +133,39 @@ public bool PreviewOrderButtonIsDisabled()
             return Driver.FindElement(Pages.OrderForm.PreviewOrderButton).FindElement(By.TagName("a")).GetAttribute("aria-disabled") != null;
         }
 
-        public bool SubmitOrderButtonIsDisabled()
+		public void SelectSupplierWithContactDetails(string bapiConnString)
+		{
+            var suppliers = ListOfSupplierNames();
+            var supplierNames = suppliers.Select(s => s.Text);
+            
+            // Randomise ordering of list so a random supplier is chosen each time
+            Random rng = new Random();
+            var supplierListShuffle = supplierNames
+                .Select(x => new { value = x, order = rng.Next() })
+                .OrderBy(x => x.order)
+                .Select(x => x.value)
+                .ToList();
+
+            string selectedSupplier = string.Empty;
+
+            foreach(var supplier in supplierListShuffle)
+			{
+                if(SupplierInfo.SupplierHasContactInfo(bapiConnString, supplier))
+				{
+                    selectedSupplier = supplier;
+                    break;
+				}
+			}
+
+            suppliers.Single(s => s.Text == selectedSupplier).Click();
```

Comment:
what if selectedSupplier  stays empty?

Reply:
added an exception if it's null/empty after loop